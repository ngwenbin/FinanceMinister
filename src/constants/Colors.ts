const primary = {
  100: "#CCE1FE",
  200: "#9AC1FE",
  300: "#689EFD",
  500: "#054EFA",
  700: "#022CB3",
  800: "#011F90",
};

const secondary = {
  200: "#FFECB1",
  500: "#FFBC3C",
  900: "#7A440B",
};

const gray = {
  50: "#F9FAFB",
  100: "#F3F4F6",
  200: "#E5E7EB",
  300: "#D1D5DB",
  400: "#9CA3AF",
  500: "#6B7280",
  600: "#4B5563",
  700: "#374151",
  800: "#1F2937",
  900: "#111827",
};

const error = {
  100: "#FFE7D3",
  200: "#FFC9A6",
  500: "#FF4823",
  700: "#B71411",
  900: "#7A0616",
};

const warning = {
  100: "#FFF8CF",
  200: "#FFF09F",
  500: "#FFCB11",
  700: "#B78908",
  900: "#7A5503",
};

const success = {
  100: "#EEFCD1",
  200: "#DAFAA5",
  500: "#72D11F",
  700: "#40960F",
  900: "#1D6405",
};

const info = {
  100: "#DDEFFE",
  200: "#BDDEFE",
  500: "#5997F9",
  700: "#2C56B3",
  900: "#112877",
};

export const Colors = {
  primary,
  secondary,
  gray,
  error,
  warning,
  success,
  info,
  white: "#FFFFFF",
  black: "#000000",
};
