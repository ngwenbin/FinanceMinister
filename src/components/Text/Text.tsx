import {
  Text as DefaultText,
  TextProps as DefaultTextProps,
} from "react-native";

import type { TextVariants } from "@constants/Fonts";
import { TextVariantsData } from "@constants/Fonts";
import { getFont } from "@utils";
import { testIds } from "@utils/tests/testIds";
import { Colors } from "@constants/Colors";

interface TextProps extends DefaultTextProps {
  children: React.ReactNode;
  variant?: TextVariants;
  color?: string;
}

const Text = ({
  children,
  color = Colors.gray[900],
  variant = "body-regular",
  ...props
}: TextProps) => {
  const { variation, size, customStyle } = TextVariantsData[variant];

  return (
    <DefaultText
      testID={testIds.sharedComponents.text}
      {...props}
      style={[{ color }, getFont(variation, size, customStyle), props.style]}
    >
      {children}
    </DefaultText>
  );
};

export default Text;
