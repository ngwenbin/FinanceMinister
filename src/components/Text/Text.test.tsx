import { render, screen } from "@testing-library/react-native";
import "@testing-library/jest-native/extend-expect";

import Text from "./Text";
import { testIds } from "@utils/tests/testIds";
import { Colors } from "@constants/Colors";

const textContent = "This is a text";
const textTestId = testIds.sharedComponents.text;

describe("<Button>", () => {
  describe("as default", () => {
    test("Renders successfully", () => {
      const activeText = render(<Text>{textContent}</Text>);
      expect(activeText).toBeDefined();
    });

    test("Renders correctly", () => {
      const activeText = render(<Text>{textContent}</Text>);
      expect(activeText.toJSON()).toMatchSnapshot();
    });

    test("Outputs the correct text", () => {
      render(<Text>{textContent}</Text>);
      expect(screen.getByTestId(textTestId)).toHaveTextContent(textContent);
    });

    test("Outputs the correct text color", () => {
      render(<Text color={Colors.primary[500]}>{textContent}</Text>);
      expect(screen.getByTestId(textTestId)).toHaveStyle({
        color: Colors.primary[500],
      });
    });
  });
});
