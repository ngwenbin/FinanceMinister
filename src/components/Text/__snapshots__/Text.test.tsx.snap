// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text> as default Renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": undefined,
      },
      Object {
        "fontFamily": "Inter_400Regular",
        "fontSize": 12,
      },
      undefined,
    ]
  }
  testID="text-0"
>
  This is a text
</Text>
`;
